### PR TITLE
remove the deprecated functions/reduce the number of disabled clippy linters

### PR DESCRIPTION
**Pull request by bilelmoussaoui**

### CODE REVIEW

File: gtk/src/auto/enums.rs [hunk 1]
```diff
@@ -7686,116 +7506,6 @@ impl ToValue for StackTransitionType {
     }
 }
 
-#[deprecated = "Since 3.14"]
```
Comment (sdroege) [inline gtk/src/auto/enums.rs:7689]: Should maybe update the minimum supported version before this release... 3.14 is from 2014. That's ancient :) Maybe 3.20 would be a good next step, that's 2016 so would still cover Ubuntu 16.04 for example.

@GuillaumeGomez @bilelmoussaoui What do you think? That would be a separate PR of course.

File: graphene/src/frustum.rs [hunk 2]
```diff
@@ -10,7 +10,7 @@ impl Frustum {
     #[doc(alias = "get_planes")]
     pub fn planes(&self) -> [Plane; 6] {
         unsafe {
-            let mut out: [ffi::graphene_plane_t; 6] = std::mem::uninitialized();
+            let mut out: [ffi::graphene_plane_t; 6] = std::mem::zeroed();
```
Comment (sdroege) [inline graphene/src/frustum.rs:13]: Yeah it's a bit annoying that `MaybeUninit` doesn't work nicely for arrays yet.

File: gdk/src/lib.rs [hunk 3]
```diff
@@ -3,7 +3,6 @@
 #![cfg_attr(feature = "dox", feature(doc_cfg))]
 #![allow(clippy::missing_safety_doc)]
 #![allow(clippy::upper_case_acronyms)]
-#![allow(deprecated)]
```
Comment (sdroege) [inline gdk/src/lib.rs:6]: Because of this we get many warnings like

```
use of deprecated static `auto::constants::STYLE_REGION_TAB`: Since 3.20
```

now. Which is of course correct but also... why would it warn about that inside the crate that defines these things? :) Somehow you need to keep the deprecated API alive, and that requires using it inside the crate.

What should we should do about that?

**Review by sdroege — COMMENTED** [review]
Looks good to me otherwise, nice cleanup!

File: graphene/src/box_.rs [hunk 4]
```diff
@@ -10,7 +10,7 @@ impl Box {
     #[doc(alias = "get_vertices")]
     pub fn vertices(&self) -> [Vec3; 8] {
         unsafe {
-            let mut out: [ffi::graphene_vec3_t; 8] = std::mem::uninitialized();
+            let mut out: [ffi::graphene_vec3_t; 8] = std::mem::zeroed();
```
Comment (GuillaumeGomez) [inline graphene/src/box_.rs:13]: Considering that we "fill" the array just below, is there any point in filling it with zeroes? Performance wise, it doesn't seem like a good idea...

File: gtk/src/auto/enums.rs (hunk 1 above)
Comment (GuillaumeGomez) [reply gtk/src/auto/enums.rs:7689]: Sounds good to me.

File: graphene/src/box_.rs (hunk 4 above)
Comment (sdroege) [reply graphene/src/box_.rs:13]: It's not UB like before :)

Comment (GuillaumeGomez) [reply graphene/src/box_.rs:13]: The content is all initialized, but then we call a function which might put random data, so back to square one. In my opinion, this is not an improvement in this context.

Comment (sdroege) [reply graphene/src/box_.rs:13]: There's nothing about opinions here. The old code is UB because you are not allowed to get a reference to uninitialized memory. The new code is fine because it initializes it with zeroes (which is a valid value in this case!) before creating a reference that is passed to the C function as a pointer.

What the C functions does or doesn't do is irrelevant in this context.

Comment (GuillaumeGomez) [reply graphene/src/box_.rs:13]: Now that I think about it: why aren't we using `MaybeUninit` here?

File: gdk/src/lib.rs (hunk 3 above)
Comment (bilelmoussaoui) [reply gdk/src/lib.rs:6]: That's annoying indeed, but only happens when generating docs it seems. Maybe a rustdoc issue/behaviour?

Comment (sdroege) [reply gdk/src/lib.rs:6]: Ok :)

File: graphene/src/box_.rs (hunk 4 above)
Comment (GuillaumeGomez) [reply graphene/src/box_.rs:13]: Ok nevermind, I thought it was an array of structs whereas it's an array of pointers.

File: gtk/src/auto/enums.rs (hunk 1 above)
Comment (bilelmoussaoui) [reply gtk/src/auto/enums.rs:7689]: No opinions on that, but seems like everyone has 3.20 nowadays

**Review by GuillaumeGomez — APPROVED** [review]
